tests/src/helper.rs

```diff
@@ -67,3 +78,93 @@ pub fn xy<X: Into<f64>, Y: Into<f64>>(x: X, y: Y) -> Coordinate<f64> {
         y: y.into(),
     }
 }
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum TestOperation {
+    Intersection,
+    Union,
+    Xor,
+    DifferenceAB,
+    DifferenceBA,
+}
+
+#[derive(Debug)]
+struct ExpectedResult {
+    result: MultiPolygon<f64>,
+    op: TestOperation,
+}
+
+fn extract_multi_polygon(feature: &Feature) -> MultiPolygon<f64> {
+    let geometry_value = feature.geometry.as_ref().expect("Feature must have 'geometry' property").value.clone();
+    let multi_polygon: MultiPolygon<f64> = match geometry_value {
+        Value::Polygon(_) => MultiPolygon(vec![geometry_value.try_into().unwrap()]),
+        Value::MultiPolygon(_) => geometry_value.try_into().unwrap(),
+        _ => panic!("Feature must either be MultiPolygon or Polygon"),
+    };
+    multi_polygon
+}
+
+fn extract_expected_result(feature: &Feature) -> ExpectedResult {
+    let multi_polygon = extract_multi_polygon(feature);
+
+    let op = feature.properties
+        .as_ref()
+        .expect("Feature needs 'properties'.")
+        .get("operation")
+        .expect("Feature 'properties' needs an 'operation' entry.")
+        .as_str()
+        .expect("'operation' entry must be a string.");
+
+    let op = match op {
+        "union" => TestOperation::Union,
+        "intersection" => TestOperation::Intersection,
+        "xor" => TestOperation::Xor,
+        "diff" => TestOperation::DifferenceAB,
+        "diff_ba" => TestOperation::DifferenceBA,
+        _ => panic!(format!("Invalid operation: {}", op)),
+    };
+
+    ExpectedResult{
+        result: multi_polygon,
+        op: op,
+    }
+}
+
+pub fn load_generic_test_case(filename: &str, regenerate: bool) {
```

Comment:
Function name isn't fully appropriate anymore => rename to `run_generic_test_case` or so.

Reply:
`run_generic_test_case` sounds about right